pkgs/applications/window-managers/hyprland/default.nix

```diff
@@ -0,0 +1,71 @@
+{ lib
+, stdenv
+, fetchFromGitHub
+, cmake
+, pkg-config
+, libdrm
+, libinput
+, libxcb
+, libxkbcommon
+, mesa
+, pango
+, wayland
+, wayland-protocols
+, wlroots
+, xcbutilwm
+}:
+
+stdenv.mkDerivation rec {
+  pname = "hyprland";
+  version = "0.5.0beta";
+
+  src = fetchFromGitHub {
+    owner = "vaxerski";
+    repo = pname;
+    rev = "v${version}";
+    fetchSubmodules = true;
```

Comment:
```suggestion
```
We fetch an unstable version of wlroots ourselves. Could we also use the version in nipxkgs?

Reply:
It might be worth leaving that as a comment? In addition, is there a place where future maintainers can determine the required wlroots?

Reply:
hyprland use a specific wlroots mirror on github since this [commit](https://github.com/vaxerski/Hyprland/commit/fa6530c7e8f5f79041a28456ca91d4641c7ad354)

Reply:
> Hyprland requires a very specific, usually HEAD, version of wlroots to build, the version packaged in nixpkgs doesn't work with it.

Does it not compile with our version or is functionality broken? If not I would suggest to ignore upstreams preference a newer version.

Reply:
It does not compile with any version of wlroots in nixpkgs.

Reply:
I'm already not fetching the submodule anymore and just patching hyprland to use headers from the wlroots we are providing.